spec/unit/helper_spec.rb

```diff
@@ -0,0 +1,164 @@
+
+describe 'PathHelpers' do
+  describe 'filename_from_url' do
+    let(:expected_filename) { 'my_filename.tgz' }
+    it 'should return the filename in a URL' do
+      url = 'https://cool.website.internet/place/my_filename.tgz'
+      expect(CernerSplunk::PathHelpers.filename_from_url(url)).to eq(expected_filename)
+    end
+  end
+end
+
+# Fake notifies method because we aren't using ChefSpec here.
+def notifies(*any)
+end
+
+describe 'RestartHelpers' do
+  let(:install_dir) { '/opt/splunk' }
+  let(:marker) { double('marker_path') }
+  before do
+    expect(CernerSplunk::RestartHelpers).to receive(:marker_path).and_return(marker).at_least :once
+    allow(CernerSplunk::RestartHelpers).to receive(:resources)
+  end
+
+  describe 'ensure_restart' do
+    it 'should create the restart marker' do
+      expect(marker).to receive(:exist?).and_return false
+      expect(marker).to receive(:open)
+      CernerSplunk::RestartHelpers.ensure_restart
+    end
+
+    context 'when the marker exists' do
+      it 'should not place the restart marker' do
+        expect(marker).to receive(:exist?).and_return true
+        expect(marker).not_to receive(:open)
+        expect(CernerSplunk::RestartHelpers).to receive(:notifies)
+        CernerSplunk::RestartHelpers.ensure_restart
+      end
+    end
+  end
+
+  describe 'check_restart' do
+    it 'should not restart the splunk service' do
+      expect(marker).to receive(:exist?).and_return false
+      expect(CernerSplunk::RestartHelpers).not_to receive(:notifies)
+      CernerSplunk::RestartHelpers.check_restart
+    end
+
+    context 'when the marker exists' do
+      it 'should restart the splunk service' do
+        expect(marker).to receive(:exist?).and_return true
+        expect(CernerSplunk::RestartHelpers).to receive(:notifies)
+        CernerSplunk::RestartHelpers.check_restart
+      end
+    end
+  end
+
+  describe 'clear_restart' do
+    it 'should not delete the restart marker' do
+      expect(marker).to receive(:exist?).and_return false
+      expect(marker).not_to receive(:delete)
+      CernerSplunk::RestartHelpers.clear_restart
+    end
+
+    context 'when the marker exists' do
+      it 'should delete the restart marker' do
+        expect(marker).to receive(:exist?).and_return true
+        expect(marker).to receive(:delete)
+        CernerSplunk::RestartHelpers.clear_restart
+      end
+    end
+  end
+end
+
+describe 'ResourceHelpers' do
+  describe 'load_installation_state' do
+    let(:name) { 'splunk' }
+    let(:install_dir) { '/opt/splunk' }
+    # class Node
+    #   attr_accessor :run_state
+    #   def initialize
+    #     @run_state = {}
+    #   end
+    # end
+    let(:node) do
+      node = Chef::Node.new
+      node.automatic['os'] = 'linux'
+      node.automatic['kernel'] = { 'machine' => 'x86_64' }
+      node
+    end
+    let(:install_dir_path) { double('Pathname.new(install_dir)') }
```

Comment:
Yeah, I named it to represent the object it's mocking. I could name it 'install_dir' if that makes more sense.